src/main/groovy/edu/oregonstate/mist/videogamesampleapi/resources/PlatformResource.groovy

```diff
@@ -110,4 +111,42 @@ class PlatformResource extends Resource {
             responseBuilder.build()
         }
     }
+
+    @Path("/{id : \\d+}")
+    @PUT
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response putPlatform(@Valid Platform newPlatform, @PathParam("id") Integer id) {
+        ResponseBuilder responseBuilder
+        Platform returnPlatform
+
+        Platform checkPlatform = platformDAO.getPlatformById(id)
+
+        //POSTs if the platform doesn't already exist
+        if (checkPlatform == null) {
+            try {
+                platformDAO.postPlatform(newPlatform.getReleaseYear(), newPlatform.getName(), newPlatform.getManufacturer(), newPlatform.getComputer(), newPlatform.getConsole())
```

Comment:
In groovy you can use: newPlatform.releaseYear, newPlatform.name, ...
This calls the getter methods in your groovy object. It will make this line shorter and more readable.